trickops.py

```diff
@@ -0,0 +1,42 @@
+import sys
+import os
+
+sys.path.append(sys.argv[1] + "/share/trick/trickops")
+# print (sys.path)
+from TrickWorkflow import *
+class SimTestWorkflow(TrickWorkflow):
+    def __init__( self, quiet, trick_top_level ):
+        # Create the trick_test directory if it doesn't already exist
+        if not os.path.exists(trick_top_level + "/trick_test"):
+          os.makedirs(trick_top_level + "/trick_test")
+
+        # Base Class initialize, this creates internal management structures
+        TrickWorkflow.__init__(self, project_top_level=(trick_top_level), log_dir=(trick_top_level +'/trickops_logs/'),
+            trick_dir=trick_top_level, config_file=(trick_top_level + "/test_sims.yml"), cpus=3, quiet=True)
+    def run( self ):
+
+      # Change all the build commands to have the whole path
```

Comment:
I ran into something similar yesterday when I realized that `${TRICK_HOME}/bin/` has to be on your `PATH` or sim builds will fail with `trick-CP: command not found`. Never ran into this in my Gateway usage of TrickOps because our project environment takes care of it.  Here's what I did as a quick hack, what do you think?

```diff
diff --git a/share/trick/trickops/TrickWorkflow.py b/share/trick/trickops/TrickWorkflow.py
index 8ab5076..4ebb535 100644
--- a/share/trick/trickops/TrickWorkflow.py
+++ b/share/trick/trickops/TrickWorkflow.py
@@ -100,9 +100,6 @@ class TrickWorkflow(WorkflowCommon):
             available
         """
         super().__init__(project_top_level=project_top_level, log_dir=log_dir, quiet=quiet)
-        # If not found in the config file, these defaults are used
-        self.defaults = {'cpus': 3, 'name': None, 'description': None,
-          'build_command': 'trick-CP', 'size': 2200000, 'env': None}
         self.config_errors = False
         self.compare_errors = False     # True if comparison errors were found
         self.sims = []                  # list of Sim() instances, filled out from config file
@@ -113,6 +110,9 @@ class TrickWorkflow(WorkflowCommon):
         self.config = self._read_config(self.config_file) # Contains resulting Dict parsed by yaml.load
         self.trick_dir = trick_dir
         self.trick_host_cpu = self.get_trick_host_cpu()
+        # If not found in the config file, these defaults are used
+        self.defaults = {'cpus': 3, 'name': None, 'description': None,
+          'build_command': os.path.join(self.trick_dir,'bin/trick-CP'), 'size': 2200000, 'env': None}
         self._validate_config()
 
     def create_test_suite(self):
```

Reply:
I like the idea of having TrickOps automatically append the whole path better than what I ended up doing, but this particular snippet of code wouldn't work for me since I have to use 'trick-CP -t' and this only sets the defaults. Maybe either have a flag to use the absolute path, or just do it automatically? I can't think of a case where you would want to use a build command that isn't just arguments to trick-CP

Reply:
All calls to all trick utilties, trick-CP, trick-ICG, trick-dp, etc.  should be made with the full path.  Otherwise you are vulnerable to users having the wrong environment and trick-CP can be called from the wrong Trick directory.

Reply:
This applies to all scripts, makefiles, system calls in code, everything.

Reply:
Agreed. I can only think of `trick-CP <args>` as the use case for specifying `build_command`, so I like your suggestion of just tacking on the full absolute path automatically. I probably should have done it this way to begin with, oversight on my part.

Reply:
It's a lesson I've learned many times. 🤕